src/htmdec_formats/keyence_formats.py

```diff
@@ -0,0 +1,251 @@
+import base64
+import io
+import os
+import pathlib
+import sys
+import xml.etree.ElementTree as ET
+import zipfile
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xlsxwriter
+from matplotlib.colors import ListedColormap
+
+try:
+    from .ksy_files.dotnetlist import Dotnetlist
+    from .ksy_files.vk4 import Vk4
+except ImportError as e:
+    print(f"Error: {e}")
+    print("Please run 'make install' to compile missing modules")
+    sys.exit(1)
+
+
+class CAGDataset:
+    def __init__(self, filename: str):
+        self.filename = filename
+        self.meta_files = {}
+        self.volumes = {}
+        self.measurements = {}
+        self.vk6_files = {}
+
+        # CAG uses a lot of uuid instead of human readable names
+        # following mapping is used to map uuid to something more reasonable.
+        # Note that names are made up.
+        self.fmapping = {
+            "f1724dc6-686c-4502-9227-2a594bc8ed33": "index.xml",
```

Comment:
names are made up, or uuids are made up?  Are you sure that these UUIDs are general?

Reply:
names are made up. UUIDs were the same in every file I checked.